core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -97,6 +97,49 @@ package object config {
     .toSequence
     .createWithDefault(Nil)
 
+  // Blacklist confs
+  private[spark] val BLACKLIST_ENABLED =
+    ConfigBuilder("spark.blacklist.enabled")
+    .booleanConf
+    .createOptional
+
+  private[spark] val MAX_TASK_ATTEMPTS_PER_NODE =
+    ConfigBuilder("spark.blacklist.maxTaskAttemptsPerNode")
+    .intConf
+    .createWithDefault(2)
+
+  private[spark] val MAX_FAILURES_PER_EXEC =
+    ConfigBuilder("spark.blacklist.maxFailedTasksPerExecutor")
+    .intConf
+    .createWithDefault(2)
+
+  private[spark] val MAX_FAILURES_PER_EXEC_STAGE =
+    ConfigBuilder("spark.blacklist.maxFailedTasksPerExecutorStage")
+    .intConf
+    .createWithDefault(2)
+
+  private[spark] val MAX_FAILED_EXEC_PER_NODE =
+    ConfigBuilder("spark.blacklist.maxFailedExecutorsPerNode")
+    .intConf
+    .createWithDefault(2)
+
+  private[spark] val MAX_FAILED_EXEC_PER_NODE_STAGE =
+    ConfigBuilder("spark.blacklist.maxFailedExecutorsPerNodeStage")
+    .intConf
+    .createWithDefault(2)
+
+  private[spark] val BLACKLIST_EXPIRY_TIMEOUT_CONF =
+    ConfigBuilder("spark.scheduler.blacklist.recoverPeriod")
```

Comment:
Do you think timeout is more clear here? (spark.scheduler.blacklist.timeout) or expiryTimeout or expireTimeout?  "recover" can have various meanings -- and we're not actively recovering the machine, or doing any test of whether it's recovered, so I think the simpler semantics of timeout or expiryTimeout might be more clear.